cmake/kokkos_arch.cmake

```diff
@@ -30,24 +30,40 @@ SET(KOKKOS_ARCH_LIST)
 
 
 KOKKOS_DEPRECATED_LIST(ARCH ARCH)
-KOKKOS_ARCH_OPTION(NATIVE          HOST "local machine")
-KOKKOS_ARCH_OPTION(AMDAVX          HOST "AMD chip")
-KOKKOS_ARCH_OPTION(ARMV80          HOST "ARMv8.0 Compatible CPU")
-KOKKOS_ARCH_OPTION(ARMV81          HOST "ARMv8.1 Compatible CPU")
-KOKKOS_ARCH_OPTION(ARMV8_THUNDERX  HOST "ARMv8 Cavium ThunderX CPU")
-KOKKOS_ARCH_OPTION(ARMV8_THUNDERX2 HOST "ARMv8 Cavium ThunderX2 CPU")
-KOKKOS_ARCH_OPTION(A64FX           HOST "ARMv8.2 with SVE Support")
-KOKKOS_ARCH_OPTION(WSM             HOST "Intel Westmere CPU")
-KOKKOS_ARCH_OPTION(SNB             HOST "Intel Sandy/Ivy Bridge CPUs")
-KOKKOS_ARCH_OPTION(HSW             HOST "Intel Haswell CPUs")
-KOKKOS_ARCH_OPTION(BDW             HOST "Intel Broadwell Xeon E-class CPUs")
-KOKKOS_ARCH_OPTION(SKX             HOST "Intel Sky Lake Xeon E-class HPC CPUs (AVX512)")
-KOKKOS_ARCH_OPTION(KNC             HOST "Intel Knights Corner Xeon Phi")
-KOKKOS_ARCH_OPTION(KNL             HOST "Intel Knights Landing Xeon Phi")
-KOKKOS_ARCH_OPTION(BGQ             HOST "IBM Blue Gene Q")
-KOKKOS_ARCH_OPTION(POWER7          HOST "IBM POWER7 CPUs")
-KOKKOS_ARCH_OPTION(POWER8          HOST "IBM POWER8 CPUs")
-KOKKOS_ARCH_OPTION(POWER9          HOST "IBM POWER9 CPUs")
+
+SET(HOST_ARCH_ALREADY_SPECIFIED "")
+MACRO(CHECK_HOST_ARCH ARCH LABEL)
```

Comment:
I would rename this to make it clearer that this also declares the option since it's easily missed

Maybe something like `DECLARE_AND_CHECK_HOST_ARCH` though that's kinda verbose